full-stack-practice/client/src/lib/read.ts

```diff
@@ -0,0 +1,29 @@
+import { type Product } from './data';
+
+export async function readCatalog(): Promise<Product[]> {
+  try {
```

Comment:
It's best to not have try/catch in these functions. The try/catch will be in the React component that calls these functions. (This applies to both functions.)